plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -575,6 +630,40 @@
     node: sabrewulf
     protection: yes
 
+- name: Create new VM using cloud-init with a username and password
+  community.general.proxmox_kvm:
+    node: sabrewulf
+    api_user: root@pam
+    api_password: secret
+    api_host: helldorado
+    name: spynal
+    ide: '{ "ide2": "local:cloudinit,format=qcow2"}'
+    ciuser: mylinuxuser
+    cipassword: supersecret
+    searchdomains: "mydomain.internal"
+    nameservers: 1.1.1.1
+    net: '{"net0":"virtio,bridge=vmbr1,tag=77"}'
+    ipconfig:
+      0:
+        ip: '192.168.1.1/24'
+        gw: '192.168.1.1'
```

Comment:
The `ipconfig` parameter should be presented the same way as the other dicts, or else it will not be transmitted to the API by `create_vm()`.

Example:

The following parameters are fed in a task to the `proxmox_kvm` module:

```yaml
net:
  net0: "virtio,bridge=vmbr0"
  net1: "virtio,bridge=vmbr1"
ide:
  ide2: "local:cloudinit,format=qcow2"
ipconfig:
  0:
    ip: "192.168.1.1/24"
```

The function `create_vm()` transforms them by extracting sub dictionaries into the main object, removing any `None`, converting booleans, etc...
In the end our initial parameters look like this:

```json
{
  "net0": "virtio,bridge=vmbr0",
  "net1": "virtio,bridge=vmbr1",
  "ide2": "local:cloudinit,format=qcow2",
  "0": {
    "ip": "192.168.1.1/24"
  }
}
```

The element "0" is then rejected by Proxmox in my tests.

```suggestion
    ipconfig:
      ipconfig0: "ip=192.168.1.1/24,gw=192.168.1.1"
```

With this fix my VM was properly configured with a cloud-init drive and net0 has IP address 192.168.1.1/24. It is not as beautiful perhaps but your method of splitting parameters can be added later :)

Reply:
you're right, i've updated the docs already but forgot about the examples ;)